task.ps1

```diff
@@ -21,3 +22,28 @@ $nsgRuleHTTP = New-AzNetworkSecurityRuleConfig -Name HTTP  -Protocol Tcp -Direct
 New-AzNetworkSecurityGroup -Name $networkSecurityGroupName -ResourceGroupName $resourceGroupName -Location $location -SecurityRules $nsgRuleSSH, $nsgRuleHTTP
 
 # ↓↓↓ Write your code here ↓↓↓
+Write-Host "Creating a virtual network $virtualNetworkName and a subnet $subnetName ..."
+New-AzVirtualNetwork -Name $virtualNetworkName -ResourceGroupName $resourceGroupName -Location $location -AddressPrefix $vnetAddressPrefix -Subnet @(New-AzVirtualNetworkSubnetConfig -Name $subnetName -AddressPrefix $subnetAddressPrefix)
+
+Write-Host "Creating a public IP address $publicIpAddressName with DNS label ..."
+New-AzPublicIpAddress -ResourceGroupName $resourceGroupName -Location $location -Name $publicIpAddressName -AllocationMethod Static -Sku Basic -DomainNameLabel $domainNameLabel
+
+Write-Host "Creating a SSH key resource $sshKeyName ..."
+New-AzSshKey -ResourceGroupName $resourceGroupName -Name $sshKeyName -PublicKey "$sshKeyPublicKey"
+
+Write-Host "Creating a virtual machine $vmName ..."
+$credential = Get-Credential
```

Comment:
The `Get-Credential` command prompts for credentials. Ensure that this is handled appropriately in automated environments, as it requires user interaction.